### ISSUE TITLE
Too easy to write a refactoring test that doesn't validate anything

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn-sdk#991]
Recently converting an analyzer/codefix pair to a refactoring, and found that it was very easy to get into a state where I had a test to check that the refactoring wasn't offered, that passed, but didn't actually validate anything.

eg, this test, when executed against an analyzer, verifies that no diagnostic is reported. The same test, against a refactoring provider, doesn't verify anything, not even run any of the code in the refactoring provider.

```cs
            await new VerifyCS.Test
            {
                TestCode = @"
public class C
{
    public void M()
    {
        var x = new byte[] { 65, 66, 67 };
    }
}"
            }.RunAsync();
```

It would be good if the test could fail if there is no cursor location or selection span in the input (it does if you provide a FixedCode), and if there is a span present it would be good if it verified that no refactoring was offered.

I might be doing something completely wrong in writing these tests, but without the test failing its very easy to miss that, and have no idea how to fix it anyway.
